feat: add save model to t2b

gurdeep330 opened the pull request:
# For authors

## Description

In this PR, I propose a new tool `save_tool`, which will save the model saved in the key `model_as_string`. The tool `simulate_model` will by default save the simulated model in the key `model_as_string` as string.

## Fixes # NA

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?
Please describe the tests you conducted to verify your changes. These may involve creating new test scripts or updating existing ones.
- [X] Added new test(s) in the ```tests``` folder
- [x] Added new function(s) to an existing test(s) (e.g.: ```tests/testX.py```)
- [ ] No new tests added (Please explain the rationale in this case)

## Checklist
- [X] My code follows the style guidelines mentioned in the Code/DevOps guides
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (e.g. MkDocs)
- [X] My changes generate no new warnings
- [X] I have added or updated tests (in the ```tests``` folder) that prove my fix is effective or that my feature works
- [X] New and existing tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

# For reviewers

## Checklist pre-approval
- [ ] Is there enough documentation?
- [ ] If a new feature has been added, or a bug fixed, has a test been added to confirm good behavior?
- [ ] Does the test(s) successfully test edge/corner cases?
- [ ] Does the PR pass the tests? (if the repository has continuous integration)

## Checklist post-approval
- [ ] Does this PR merge ```develop``` into ```main```? If so, please make sure to add a prefix (feat/fix/chore) and/or a suffix BREAKING CHANGE (if it's a major release) to your commit message.
- [ ] Does this PR close an issue? If so, please make sure to descriptively close this issue when the PR is merged.

## Checklist post-merge
- [ ] When you approve of the PR, merge and close it (Read this [article](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/about-merge-methods-on-github) to know about different merge methods on GitHub)
- [ ] Did this PR merge ```develop``` into ```main``` and is it suppose to run an automated release workflow (if applicable)? If so, please make sure to check under the "Actions" tab to see if the workflow has been initiated, and return later to verify that it has completed successfully.

lilijap reviewed (changes requested):
Hi @gurdeep330!

Thanks so much for adding this tool! It’s working well on my side, though the file name is fixed to `saved_model.xml`. Maybe you could add a short note about this limitation and the note that exact path has to be specified in the prompt, in the tool’s description page on the website and on the welcome page?

<img width="846" height="685" alt="image" src="https://github.com/user-attachments/assets/7b914f08-5cbc-4c79-bb48-21ad018aa5c6" />

Thank you,
Lilija

gurdeep330 commented:
> Hi @gurdeep330!
> 
> Thanks so much for adding this tool! It’s working well on my side, though the file name is fixed to `saved_model.xml` and should not be specified in the path, when saving the model. Maybe you could add a short note about this limitation and the note that exact path has to be specified in the prompt, in the tool’s description page on the website and on the welcome page?
> 
> <img alt="image" width="846" height="685" src="https://private-user-images.githubusercontent.com/43205236/494755589-7b914f08-5cbc-4c79-bb48-21ad018aa5c6.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTkwNTMyNjAsIm5iZiI6MTc1OTA1Mjk2MCwicGF0aCI6Ii80MzIwNTIzNi80OTQ3NTU1ODktN2I5MTRmMDgtNWNiYy00Yzc5LWJiNDgtMjFhZDAxOGFhNWM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTI4VDA5NDkyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc0M2RhYTA4OGM3M2M1Zjc4NzYwYzI2MjhmNzI5ZDg3ZTk0Y2Q0Njg5MWY1NTA0ZjVhNTdmODBhNmZjMzBhZjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.fjUcPZAl1N_umJS7mSdDD81_h1hl_HRTPhE2q3I4jlc">
> Thank you, Lilija

Hi @lilijap 

I have updated the save model tool:
 - It takes input as both the path to the output file and the output file name.
 - The path is set to '.' (current working directory), and the output file name is set to 'saved_model.xml' as the default.
 - The user can also overwrite these defaults by specifying these arguments in the prompt.
 - The agent will also mention the path and the output file name in its response.

I have also updated the prompt to specify the tool names and descriptions in the starter message. The LLM generates this message and, therefore, may not always contain the arguments. However, the user can always prompt the agent to ask for information on any tool. Additionally, I have also updated the tool description in the code to generate the webpage.

Let me know if this addresses your concerns. Thx!
